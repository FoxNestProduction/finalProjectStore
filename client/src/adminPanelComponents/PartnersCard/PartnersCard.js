import { Box, Card, CardHeader, Button, CardMedia, Typography } from '@mui/material';
import React from 'react';
import PropTypes from 'prop-types';

const PartnersCard = ({ title, url, enabled }) => {
  return (
    <Box>
      <Card
        sx={{
          maxWidth: '1083px',
          maxHeight: '611px',
          width: '100%',
          padding: '10px 24px 24px 24px',
        }}
      >
        <CardHeader
          sx={{
            p: '0 16px 16px 16px',
            position: 'relative',
          }}
          title={(
            <Typography
              sx={{
                fontSize: {
                  mobile: '38px',
                  fontWeight: '500',
                  fontFamily: 'fontPoppins',
                },
              }}
            >
              {title}
            </Typography>
          )}
          action={(
            <Button
              onClick={(event) => {
                event.preventDefault();
              }}
              aria-label="disable"
              variant="outlined"
              sx={{
                border: 'none',
                outline: '2px solid #FB471D',
                color: '#000000',
                fontFamily: 'fontFamily',
                fontSize: {
                  mobile: '18px',
                },
                px: '1.5em',
                position: 'absolute',
                top: '14px',
                right: '0',
                transition: 'background-color 0.3s ease, box-shadow 0.3s ease',
                '&:hover': {
                  boxShadow: '0px 0px 10px rgba(0, 0, 0, 0.3)',
                  backgroundColor: 'common.white',
                  outline: 'none',
                  border: 'none',
                },
                '&:active': {
                  boxShadow: '0px 1px 5px rgba(0, 0, 0, 0.2)',
                  transform: 'translateY(1px)',
                  backgroundColor: '#EAEAEA',
                },
              }}
            >
              Disable
            </Button>
          )}
        />
        <Box
          sx={{
            position: 'relative',
          }}
        >
          {!enabled ? (
            <Box
              sx={{
                position: 'absolute',
                top: '0',
                left: '0',
                width: '100%',
                height: '100%',
                background: 'rgba(217, 217, 217, 0.60)',

              }}
            />
          ) : null}
          <CardMedia
            component="img"
            image={`${url}`}
            sx={{
              display: 'block',
              width: '1035px',
              height: '495px',
              borderRadius: '16px',
            }}
            alt={`Restaurant ${title}`}
          />
        </Box>
      </Card>
    </Box>
  );
};

PartnersCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string,
  enabled: PropTypes.bool.isRequired,
};

PartnersCard.defaultProps = {
  url: '',
};

export default PartnersCard;
