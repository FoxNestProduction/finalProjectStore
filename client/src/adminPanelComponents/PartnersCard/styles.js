export const card = {
  maxWidth: '1083px',
  maxHeight: '611px',
  width: '100%',
  padding: '10px 24px 24px 24px',
};
export const cardHeader = {
  p: {
    tablet: '0 0 16px',
    lgTablet: '0 16px 16px 16px',
  },
  position: 'relative',
};

export const cardTitle = {
  fontSize: {
    mobile: '26px',
    tablet: '28px',
    lgTablet: '30px',
    desktop: '38px',
  },
  fontWeight: '500',
  fontFamily: 'fontPoppins',
};

export const disableButton = {
  border: 'none',
  outline: '2px solid #FB471D',
  color: '#000000',
  fontFamily: 'fontFamily',
  fontSize: {
    mobile: '18px',
  },
  px: {
    tablet: '',
    lgTablet: '1em',
    desktop: '1.5em',
  },
  position: {
    tablet: '',
    lgTablet: 'absolute',
    desktop: 'absolute',
  },
  top: '14px',
  right: '0',
  transition: 'background-color 0.3s ease, box-shadow 0.3s ease',
  '&:hover': {
    boxShadow: '0px 0px 10px rgba(0, 0, 0, 0.3)',
    backgroundColor: 'common.white',
    outline: 'none',
    border: 'none',
  },
  '&:active': {
    boxShadow: '0px 1px 5px rgba(0, 0, 0, 0.2)',
    transform: 'translateY(1px)',
    backgroundColor: '#EAEAEA',
  },
};

export const imgWrapper = {
  position: 'relative',
};

export const disabledCardStyles = {
  position: 'absolute',
  top: '0',
  left: '0',
  width: '100%',
  height: '100%',
  background: 'rgba(217, 217, 217, 0.60)',
  borderRadius: '16px',
};

export const imgStyles = {
  display: 'block',
  width: '1035px',
  height: '495px',
  borderRadius: '16px',
};
