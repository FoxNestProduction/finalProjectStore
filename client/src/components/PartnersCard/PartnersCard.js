import React from 'react';
import PropTypes from 'prop-types';
import Container from '@mui/material/Container';
import { fixedDecodeURIComponent } from '../../utils/uriEncodeHelpers';
import RestaurantCard from '../RestaurantCard/RestaurantCard';
import useGetPartners from '../../customHooks/useGetPartners';

const PartnersCard = ({ partnersName }) => {
  const partners = useGetPartners();

  const nameOfPartners = fixedDecodeURIComponent(partnersName);
  const rest = partners.find((item) => item.name.toLowerCase() === nameOfPartners);
  const styleRestaurantCard = {
    mobile: 315,
    lgTablet: 881,
    desktop: 1285,
  };

  return (
    <Container
      component="section"
      sx={{
        bgcolor: 'background.default',
        mt: { mobile: 5, tablet: 8 },
        mb: { mobile: 5, tablet: 8 },
      }}
    >

      <RestaurantCard {...rest} styleWidth={styleRestaurantCard} />
    </Container>
  );
};

PartnersCard.propTypes = {
  partnersName: PropTypes.string.isRequired,
};

export default PartnersCard;
