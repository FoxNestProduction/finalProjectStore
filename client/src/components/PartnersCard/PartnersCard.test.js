import React from 'react';
import { render } from '@testing-library/react';
import PartnersCard from './PartnersCard';

describe('ProductCardItem Component', () => {
  test('should renders ProductCardItem component', () => {
    const { asFragment } = render(
      <PartnersCard partner="partner" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
