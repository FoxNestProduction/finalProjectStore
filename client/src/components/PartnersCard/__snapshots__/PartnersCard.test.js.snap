// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCardItem Component should renders ProductCardItem component 1`] = `
<DocumentFragment>
  <section
    class="MuiContainer-root MuiContainer-maxWidthLg css-2zxudu-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-10vyi2c-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-gpemya"
      >
        <img
          alt=""
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-fh02g9-MuiCardMedia-root"
        />
      </div>
      <div
        class="MuiStack-root css-rwxpvi-MuiStack-root"
      >
        <div
          class="MuiCardHeader-root css-4mdd9q-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 css-xz6qnp-MuiTypography-root"
            />
          </div>
        </div>
        <div
          class="MuiCardContent-root css-oq62jt-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <span
              aria-label="3 Stars"
              class="MuiRating-root MuiRating-sizeSmall Mui-readOnly MuiRating-readOnly css-1nmdjm5-MuiRating-root"
              role="img"
            >
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                    data-testid="StarRateRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                    data-testid="StarRateRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                    data-testid="StarRateRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarRateRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                    />
                  </svg>
                </span>
              </span>
              <span>
                <span
                  class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarRateRoundedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m14.43 10-1.47-4.84c-.29-.95-1.63-.95-1.91 0L9.57 10H5.12c-.97 0-1.37 1.25-.58 1.81l3.64 2.6-1.43 4.61c-.29.93.79 1.68 1.56 1.09l3.69-2.8 3.69 2.81c.77.59 1.85-.16 1.56-1.09l-1.43-4.61 3.64-2.6c.79-.57.39-1.81-.58-1.81h-4.45z"
                    />
                  </svg>
                </span>
              </span>
            </span>
          </div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-w9h7d4-MuiTypography-root"
          >
            24min â€¢
          </p>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-mkkqzv-MuiCardActions-root"
        >
          <div
            class="MuiBox-root css-1ejqzbj"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
              data-testid="VerifiedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m23 12-2.44-2.79.34-3.69-3.61-.82-1.89-3.2L12 2.96 8.6 1.5 6.71 4.69 3.1 5.5l.34 3.7L1 12l2.44 2.79-.34 3.7 3.61.82L8.6 22.5l3.4-1.47 3.4 1.46 1.89-3.19 3.61-.82-.34-3.69L23 12zm-12.91 4.72-3.8-3.81 1.48-1.48 2.32 2.33 5.85-5.87 1.48 1.48-7.33 7.35z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="MuiStack-root css-iaetsx-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-5jihz5-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Show more
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <p
        class="MuiTypography-root MuiTypography-description css-9psp93-MuiTypography-root"
      />
    </div>
  </section>
</DocumentFragment>
`;
